pkg/syncer/metadatasyncer.go

```diff
@@ -302,88 +284,47 @@ func InitMetadataSyncer(ctx context.Context, clusterFlavor cnstypes.CnsClusterFl
+		// for other features instead of restarting CSI container, if possible we can implement init() function and
+		// it can initialize required things when capability value changes from false to true.
+		IsWorkloadDomainIsolationSupported = commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx,
+			common.WorkloadDomainIsolation)
 		if !IsWorkloadDomainIsolationSupported {
-			// If the WCP capability for WorkloadDomainIsolation feature is disabled,
-			// start an informer to know when the capability is enabled and restart the container.
-			err = k8s.NewConfigMapListener(
-				ctx,
-				k8sClient,
-				common.KubeSystemNamespace,
-				// Add.
-				nil,
-				// Update.
-				func(oldObj interface{}, newObj interface{}) {
-					_, log := logger.GetNewContextWithLogger()
-					oldFssConfigMap, ok := oldObj.(*v1.ConfigMap)
-					if oldFssConfigMap == nil || !ok {
-						log.Warnf("configMapUpdated: unrecognized old object %+v", oldObj)
-						return
-					}
-					newFssConfigMap, ok := newObj.(*v1.ConfigMap)
-					if newFssConfigMap == nil || !ok {
-						log.Warnf("configMapUpdated: unrecognized new object %+v", newObj)
-						return
-					}
-					if oldFssConfigMap.Name == common.WCPCapabilityConfigMapName {
-						log.Infof("Observed a change in WCP capabilities...")
-						oldFSSValue := parseBool(oldFssConfigMap, common.WorkloadDomainIsolation,
-							common.KubeSystemNamespace)
-						newFSSValue := parseBool(newFssConfigMap, common.WorkloadDomainIsolation,
-							common.KubeSystemNamespace)
-						if !oldFSSValue && newFSSValue {
-							log.Infof("%s capability is enabled in %s configmap in %s namespace. "+
-								"Restarting the container as capabilities have changed.",
-								common.WorkloadDomainIsolation, common.WCPCapabilityConfigMapName,
-								common.KubeSystemNamespace)
-							os.Exit(1)
-						}
-					}
-				},
-				// Delete.
-				nil)
-			if err != nil {
-				return logger.LogNewErrorf(log, "failed to listen on configmaps in namespace %q. Error: %v",
-					common.KubeSystemNamespace, err)
-			}
-
-			// In case the informer above missed the update event where WCP capability for
-			// WorkloadDomainIsolation feature is turned on, we will check the WCP capability configmap
-			// every 2mins to check if there is a change and restart the container if it is enabled.
+			// Check the WCP capabilities CR on supervisor cluster after every 2 mins to check if there
+			// is a change in capability value from false to true. If so, restart the CSI container.
 			go func() {
 				ticker := time.NewTicker(time.Duration(2) * time.Minute)
 				defer ticker.Stop()
+				restConfig, err := config.GetConfig()
+				if err != nil {
+					log.Errorf("failed to get Kubernetes config. Err: %+v", err)
+					os.Exit(1)
+				}
+				wcpCapabilityApiClient, err := k8s.NewClientForGroup(ctx, restConfig, wcpcapapis.GroupName)
+				if err != nil {
+					log.Errorf("failed to create wcpCapabilityApi client. Err: %+v", err)
+					os.Exit(1)
```

Comment:
if we have older supervisor where WCP capability CR is not present and TKR is latest, we will not be able to create wcpCapabilityApiClient right.
Do we want to check if type is not registered in the supervisor and handle this case?
Otherwise syncer in PVCSI will go in CLBO.